mlir/lib/Dialect/SCF/Utils/Utils.cpp

```diff
@@ -970,3 +970,69 @@ scf::ForallOp mlir::fuseIndependentSiblingForallLoops(scf::ForallOp target,
 
   return fusedLoop;
 }
+
+scf::ForOp mlir::fuseIndependentSiblingForLoops(scf::ForOp target,
+                                                scf::ForOp source,
+                                                RewriterBase &rewriter) {
+  // Create fused init_args.
+  auto targetInitArgs = target.getInitArgs();
+  auto sourceInitArgs = source.getInitArgs();
+  SmallVector<Value> fusedInitArgs;
+  fusedInitArgs.reserve(targetInitArgs.size() + sourceInitArgs.size());
+  fusedInitArgs.append(sourceInitArgs.begin(), sourceInitArgs.end());
+  fusedInitArgs.append(targetInitArgs.begin(), targetInitArgs.end());
+
+  // Create a new scf::for op after the source loop.
+  rewriter.setInsertionPointAfter(source);
+  scf::ForOp fusedLoop = rewriter.create<scf::ForOp>(
+      source.getLoc(), source.getLowerBound(), source.getUpperBound(),
+      source.getStep(), fusedInitArgs);
```

Comment:
I vaguely remember that there was some trickiness with this builder creating the `yield` automatically when the list of init args is empty, and not creating it otherwise. Not sure if this is still in place. Could you please double-check and add a test?

Reply:
You are right, this is still the case. Have now added the guard `if (yieldResults.size())` right before creating the `scf.yield`. (When `yieldResults` is empty neither loop had `iter_args` so the newly created loop will already have its terminator.)

Have added a test as well.